game-app/game-headed/build.gradle

```diff
@@ -39,41 +39,51 @@ jar {
     }
 }
 
+def assetsUrl = "https://github.com/triplea-game/assets/releases/download/47/game_headed_assets.zip"
+def assetsZipFile = file("$projectDir/.assets/assets.zip")
+def assetsCacheDir = file("$projectDir/.assets")
+def assetsUnpackedDir = file("$projectDir/assets")
+
 clean.doFirst {
-  delete "$projectDir/.assets/"
-  delete "$projectDir/assets/"
+    delete assetsCacheDir
+    delete assetsUnpackedDir
 }
 
-task downloadAssets {
-    doLast {
-        download.run {
-            src "https://github.com/triplea-game/assets/releases/download/47/game_headed_assets.zip"
-            dest "$projectDir/.assets/assets.zip"
-            overwrite false
-        }
-        copy {
-            from zipTree(file("$projectDir/.assets/assets.zip"))
-            into "$projectDir/assets/"
-        }
-    }
+ 
+task downloadAssets(type: Download) {
+    src assetsUrl
+    dest assetsZipFile
+    overwrite false
+    // declare outputs for up-to-date checks (Download task normally already behaves well)
+    outputs.file assetsZipFile
 }
 
+task unpackAssets(type: Copy) {
+    dependsOn downloadAssets
+    from { zipTree(assetsZipFile) }
+    into assetsUnpackedDir
+    inputs.file assetsZipFile
+    outputs.dir assetsUnpackedDir
+
+    // Only run the unpack if the download task actually wrote the file:
+    onlyIf { downloadAssets.didWork || !assetsUnpackedDir.exists() }
```

Comment:
It's interesting that declaring input/outputs is not sufficient to let  gradle skip this task. Do you know why this is needed? (It is needed, right?)